src/compiler/language-service.spec.ts

```diff
@@ -197,6 +199,64 @@ export const thing: Thing = { a: 1 }`
     removeSync(fileName)
   })
 
+  it('should not do type check for the test file which is already finished type checking before', () => {
+    const tmp = tempDir('compiler')
+    const testFileName = 'src/__mocks__/unchanged-modules/main.spec.ts'
+    const testFileSrc = readFileSync(testFileName, 'utf-8')
+    const importedModuleSrc = readFileSync('src/__mocks__/unchanged-modules/main.ts', 'utf-8')
+
+    const compiler = makeCompiler({
+      jestConfig: { cache: true, cacheDirectory: tmp, testMatch: ['src/__mocks__/unchanged-modules/*.spec.ts'] },
+      tsJestConfig: { tsConfig: false },
+    })
+
+    compiler.compile(testFileSrc, testFileName)
+    logTarget.clear()
+    compiler.compile(importedModuleSrc, require.resolve('../__mocks__/unchanged-modules/main.ts'))
+
+    expect(logTarget.filteredLines(LogLevels.debug, Infinity)).toMatchInlineSnapshot(`
```

Comment:
We just need to verify logging for assertion related to https://github.com/kulshekhar/ts-jest/pull/1579/files#diff-124008e76d6a710666b8d995ffc3758fR173 (the condition which is fixed in the below comment)